scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,44 @@
+
+## Decision
+
+The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD JSON feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command listed below could be used: 
+```
+curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json
+``` 
+For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels: LOW, MEDIUM, HIGH, CRITICAL. After all json files downloaded, they will be compress to one data bundle as one single compressed file.
+
+We've chosen to base our API requests on CVSS V3 severity to ensure all data integrated into Claircore possesses valid CVSS V3 metrics. Consequently, retrieving data without CVSS V3 information becomes irrelevant.
+
+The CVSS Updater in Central will be modified to download compressed file (generated by CI mentioned above) in the Google bucket. This eliminates the need for JSON parsing in the updater. Scanner V4 will pull the data bundle from Central and populate the Matcher DB with the enrichment data.
+
+A custom enricher is essential for Scanner V4, as it will provide a mechanism to retrieve JSON data from Central, parse the enrichment details, and store them in the Scanner V4 database.  Additionally, our data will be sourced from the NVD CVE API rather than JSON feeds, so the custom enricher can accommodate the distinct data formats. This also offers greater flexibility, allowing us to extract more information from the fetched JSONs than what Claircore offers.
+
+The CVSS Updater in Central will attempt to download the data bundle from the Google bucket using exponential retries in case of partial failures, up to a maximum interval within the next loop, which happens every 4 hours by default.
+
+Should the GitHub Actions workflow (CI) experience difficulties in accessing or downloading segments of the CVSS data due to an HTTP request failure—potentially caused by the NVD API being unavailable or due to rate limiting—the affected HTTP request should be automatically retried following an exponential backoff strategy. In the event that all retry attempts fail, the workflow will be terminated, and the system will await the next scheduled update in 4 hours. The team will be promptly notified of any aborted or failed workflows via Slack.
```

Comment:
It is currently 3 hours in CI. Are we making it 4 hours now?

Also, can we have a way to trigger it manually in case we detect it's up and we don't want to wait?

Reply:
We can keep it as 3 hours.
As long as we use `workflow_dispatch` in workflow, we can manually trigger a workflow.